nca/CoreDS/TcpLikeProperties.py

```diff
@@ -317,6 +366,271 @@ def _get_first_item_str(self):
         item = self.get_first_item(self.active_dimensions)
         res_list = []
         for i, dim_name in enumerate(self.active_dimensions):
-            dim_item = item[i] if dim_name != 'methods' else MethodSet.all_methods_list[item[i]]
+            if dim_name == 'protocols':
+                dim_item = ProtocolNameResolver.get_protocol_name(item[i])
+            elif dim_name == 'methods':
+                dim_item = MethodSet.all_methods_list[item[i]]
+            else:
+                dim_item = item[i]
             res_list.append(f'{dim_name}={dim_item}')
         return '[' + ','.join(s for s in res_list) + ']'
+
+    def project_on_one_dimension(self, dim_name):
+        """
+        Build the projection of self to the given dimension
+        :param str dim_name: the given dimension
+        :return: the projection on the given dimension, having that dimension type (either IntervalSet or DFA)
+        """
+        if dim_name not in self.active_dimensions:
+            if dim_name == "src_peers" or dim_name == "dst_peers":
+                return PeerSet()
+            elif dim_name == "src_ports" or dim_name == "dst_ports":
+                return PortSet()
+            elif dim_name == "protocols":
+                return ProtocolSet()
+            elif dim_name == "methods":
+                return MethodSet()
+            else:
+                return None
```

Comment:
Actually, it should be full rather than empty. Since different dimensions have different full domain values (and we probably do not want to return here all these values), I changed it to None for all non active dimensions.
Currently, this function is used only for "src_peers" and "dst_peers" dimensions.